ext/drutil/drutil.c

```diff
@@ -439,6 +465,43 @@ drutil_opnd_mem_size_in_bytes(opnd_t memref, instr_t *inst)
         uint sz = opnd_size_in_bytes(opnd_get_size(instr_get_dst(inst, 1)));
         ASSERT(opnd_is_immed_int(instr_get_src(inst, 1)), "malformed OP_enter");
         return sz * extra_pushes;
+    } else if (inst != NULL && instr_is_xsave(inst)) {
+        /*
+         * The following is an incomplete computation of the xsave instruction's
+         * written xsave area's size. Specifically, it
+         *
+         * - Ignores the user state mask components set in edx:eax, because it is
+         *   a dynamic value at runtime. The real output size of xsave depends on
+         *   the instruction's user state mask AND the user state mask as supported
+         *   by the CPU based on XCR0.
```

Comment:
It sounds like we may want to add sthg parallel to instr_compute_address_ex() for sizes which takes in the mcontext.  I would say, file an issue on that.  If it is only xsave though and if nobody cares (not sure how important it is for most memory trace uses) it would be a lower priority "help-wanted"-labeled feature.

Reply:
Ok. For instr_compute_address_ex, sounds like it uses the passed in registers from mcontext? We can make a version that is "more accurate" using the user state component mask in edx:eax.

Reply:
Yes, you use it from a clean call and pass in the live mcontext.

Reply:
Ok. done.